### PR TITLE
chore(stack, lint): DLT-2848 flex lint and flex-to-dt-stack migration script

### PR DESCRIPTION
**Pull request by francisrupert**
# Improve flex lint and add flex-to-dt-stack migration script

flex-480-x-360-gif-jqlu8zzs1af50a7f

## :hammer_and_wrench: Type Of Change

- [x] Other (chore)

https://dialpad.atlassian.net/browse/DLT-2848

## :book: Description

Favor DtStack over chaining multiple utility classes, e.g. `class="d-d-flex d-ai-* d-jc-*"`, etc. PR establishes linting and migration option for existing code.

- **Blog post:** https://dialtone.dialpad.com/deploy-previews/pr-983/about/whats-new/posts/2025-12-2.html
- **ESLint rules:** `prefer-stack-over-flex` rule and expanded `deprecated-stack-alignment-classes` to detect flex utilities that could use DtStack
- **Migration tooling:** Created `dialtone_migrate_flex_to_stack` script with interactive CLI for automated migration
- **Documentation migration:** Ran migration script on documentation md files.
- **Vue component migration:** Ran migration script on Vue2 and Vue3 stories.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

## :crystal_ball: Next Steps

- Comms to highlight Migration Script
- Run Migration Script on Beacon
- Run Migration Script **selectively** in Design Studio projects
- Run Migration Script on **narrow** UV location

## :camera: Screenshots / GIFs

image

### CODE REVIEW

**Comment by francisrupert** [comment]
No regression from Percy. Woot.

File: apps/dialtone-documentation/docs/about/whats-new/posts/2025-12-2.md
Comment (francisrupert) [inline apps/dialtone-documentation/docs/about/whats-new/posts/2025-12-2.md:1]: Blog article: https://dialtone.dialpad.com/deploy-previews/pr-983/about/whats-new/posts/2025-12-2.html

File: apps/dialtone-documentation/docs/components/card.md [hunk 1]
```diff
@@ -22,9 +22,9 @@ storybook: https://dialtone.dialpad.com/vue/?path=/story/components-card--defaul
       <div>Content slot. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum molestie semper. Morbi finibus nulla turpis, nec molestie mi rutrum.</div>
     </div>
     <div class="d-card__footer">
-      <div class="d-d-flex d-g8">
+      <dt-stack direction="row" gap="400">
```
Comment (francisrupert) [inline apps/dialtone-documentation/docs/components/card.md:25]: I don't expect you to review every instance that was migrated, but you can skim through them to pick up on the pattern the migration script does. 

I've gone through all doc pages and confirmed them.

File: packages/dialtone-css/lib/build/js/dialtone_migrate_flex_to_stack/index.mjs
Comment (francisrupert) [inline packages/dialtone-css/lib/build/js/dialtone_migrate_flex_to_stack/index.mjs:1]: This is the **key addition** to review. Account for a wide variety of edge cases. Not only have I run it on the doc site MD files and Vue Stories, I've done a bunch of dry runs on variety of places in UV and Beacon. Yeah, it's pretty complex, but ¯\_(ツ)_/¯

File: packages/dialtone-css/lib/build/js/dialtone_migrate_flex_to_stack/examples-edge-cases.vue
Comment (francisrupert) [inline packages/dialtone-css/lib/build/js/dialtone_migrate_flex_to_stack/examples-edge-cases.vue:1]: Just a test file to stress-test with variety of representative edge cases.

File: packages/dialtone-vue2/components/badge/badge_examples.story.vue
Comment (francisrupert) [inline packages/dialtone-vue2/components/badge/badge_examples.story.vue:1]: Whole lotta these too in stories for both `components` and `recipes`. I migrated in vue3 first, and then mirrored them in vue2 afterward.

File: packages/eslint-plugin-dialtone/docs/rules/deprecated-stack-alignment-classes.md
Comment (francisrupert) [inline packages/eslint-plugin-dialtone/docs/rules/deprecated-stack-alignment-classes.md:1]: This rule was instantiated in https://github.com/dialpad/dialtone/pull/979, and this PR updates it.

File: packages/eslint-plugin-dialtone/docs/rules/prefer-stack-over-flex.md
Comment (francisrupert) [inline packages/eslint-plugin-dialtone/docs/rules/prefer-stack-over-flex.md:1]: New

File: packages/dialtone-css/lib/build/less/dialtone.less [hunk 2]
```diff
@@ -98,3 +98,26 @@
 //  --  THEMES & GLOBAL SETTINGS
 @import 'dialtone-globals';
 @import 'themes/default';
+
+//  --  DEBUG
+[data-migrate-outline] {
+  --outline-color: orangered;
+  --outline-size: var(--dt-size-200);
+  --outline-offset: calc(var(--outline-size) * -1);
```
Comment (francisrupert) [inline packages/dialtone-css/lib/build/less/dialtone.less:106]: When running the script with `--show-outline` it will add `data-migrate-outline` attribute so you can visually see the results. 

Paired with a `--remove-outline` flag.

e.g.
```html
<dt-stack class="foo" data-migrate-outline>
...
</dt-stack
```

**Review by francisrupert — COMMENTED** [review]
Highlighting specific parts to simplify review.

File: packages/dialtone-css/lib/build/js/dialtone_migrate_flex_to_stack/index.mjs
Comment (braddialpad) [reply packages/dialtone-css/lib/build/js/dialtone_migrate_flex_to_stack/index.mjs:1]: This is pretty huge but at least it's well tested and documented. If we are continuing to do more stuff like this in the future we may want to consider using [codemod](https://github.com/codemod/codemod) with [vue-metamorph](https://vue-metamorph.dev/). Would probably take some time to learn, but would be easier to understand, faster and probably less error prone.

**Review by braddialpad — APPROVED** [review]
Large change, I mostly just went through the key files. Very well documented and tested.

File: apps/dialtone-documentation/docs/.vuepress/baseComponents/BaseColor.vue [hunk 3]
```diff
@@ -1,12 +1,12 @@
 <template>
-  <aside class="d-d-flex d-fd-column">
-    <header v-if="stops.length" class="d-d-flex d-jc-space-between d-ai-end">
+  <dt-stack as="aside">
+    <dt-stack v-if="stops.length" as="header" direction="row" justify="between" class="d-ai-end">
```
Comment (jeverhart-dialpad) [inline apps/dialtone-documentation/docs/.vuepress/baseComponents/BaseColor.vue]: There are just a few `d-ai-end` and `d-ai-start` classes in these files, which may express intent (likely for `d-ai-flex-start`), but don't apply any actual style. It doesn't seem like it causes an issue here, but it being the first one made it easier to notice, so just giving it a mention.

File: packages/dialtone-css/lib/build/js/dialtone_migrate_flex_to_stack/index.mjs
Comment (francisrupert) [reply packages/dialtone-css/lib/build/js/dialtone_migrate_flex_to_stack/index.mjs:1]: Yeah, I initially considered codemod (though wasn't aware of vue-metamorph) but needed support for non-vue files, e.g. js stories, md, mdx. Just .md alone was quite numerous in doc site.

But yeah, for future possible migrations tools vue-metamorph looks great.

File: apps/dialtone-documentation/docs/.vuepress/baseComponents/BaseColor.vue (hunk 3 above)
Comment (francisrupert) [reply apps/dialtone-documentation/docs/.vuepress/baseComponents/BaseColor.vue]: Nice catch. Those CSS Utilities don't exist, and so the migration script left them alone – as it should.

I'll check those out in doc site and remove, or update to `align` prop if they actually were intended as `d-ai-flex-start|end`.

Oh, and heh, in firespotter 🤔:
* [d-ai-start](https://github.com/search?q=repo%3Adialpad%2Ffirespotter+d-ai-start&type=code)
* [d-ai-end](https://github.com/search?q=repo%3Adialpad%2Ffirespotter%20d-ai-end&type=code)

I don't dare create those CSS Utilities for fear of creating regression.
